cpp/src/groupby/sort/group_m2.cu

```diff
@@ -62,15 +64,18 @@ void compute_m2_fn(column_device_view const& values,
                    ResultType* d_result,
                    rmm::cuda_stream_view stream)
 {
-  auto const var_iter = cudf::detail::make_counting_transform_iterator(
-    size_type{0},
-    m2_transform<ResultType, decltype(values_iter)>{
-      values, values_iter, d_means, group_labels.data()});
+  auto m2_fn = m2_transform<ResultType, decltype(values_iter)>{
+    values, values_iter, d_means, group_labels.data()};
+  auto const itr = thrust::make_counting_iterator<size_type>(0);
```

Comment:
I thought the factory iterator was recommended. 
I do prefer specifying the type as you suggest here.
That is, I prefer specifying 0 as `size_type` rather than the compiler automatically identifying 0 as `int` like it would with the factory without the type.

Reply:
>  I thought the factory iterator was recommended.

Yeah, I learned this from Jake's review a while ago.

> I prefer specifying 0 as size_type rather than the compiler automatically identifying 0 as int

Sounds good to me :+1: